src/Map/src/Elements.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+/**
+ * Represents a collection of map elements.
+ *
+ * @author Sylvain Blondeau <contact@sylvainblondeau.dev>
+ */
+abstract class Elements
+{
+    protected \SplObjectStorage $elements;
+
+    public function __construct(
+        array $elements,
+    ) {
+        $this->elements = new \SplObjectStorage($elements);
+        array_map(
+            fn ($element) => $this->elements->attach($element),
+            $elements
+        );
+    }
+
+    public function add(MapElement $element): self
+    {
+        $this->elements->attach($element, $element->id ?? $this->elements->getHash($element));
+
+        return $this;
+    }
+
+    private function getElement(string $id): ?Marker
+    {
+        foreach ($this->elements as $element) {
+            if ($this->elements->offsetGet($element) === $id) {
+                return $element;
+            }
+        }
```

Comment:
It is not possible to get the `$id` when iterating on `$this->elements` with `foreach`, instead of using `$this->elements->offsetGet($element)`?